src/ReaderThemeSupportFeatures.php

```diff
@@ -385,4 +471,145 @@ public function get_relative_luminance_from_hex( $hex ) {
 		$lum = ( 0.2126 * $red ) + ( 0.7152 * $green ) + ( 0.0722 * $blue );
 		return (int) round( $lum );
 	}
+
+	/**
+	 * Checks whether a theme or its parent has a theme.json file.
+	 * Checks if `wp_get_global_settings()` exists and bail for WP < 5.9.
+	 *
+	 * Copied from `wp_theme_has_theme_json()`
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @see https://github.com/WordPress/wordpress-develop/blob/200868214a1ae0a108dac491677ba82e7541fc8d/src/wp-includes/global-styles-and-settings.php#L384
+	 *
+	 * @since 2.4.1
+	 *
+	 * @return bool False if `wp_get_global_settings()` not exists or theme.json not found, true otherwise.
+	 */
+	private function theme_has_theme_json() {
+		// wp_get_global_settings() is only available in WP 5.9+.
```

Comment:
This should defer to `wp_theme_has_theme_json()` if it exists:

```suggestion
		if ( function_exists( 'wp_theme_has_theme_json' ) ) {
			return wp_theme_has_theme_json();
		}
		// wp_get_global_settings() is only available in WP 5.9+.
```